src/index.js

```diff
@@ -104,12 +104,12 @@ const rules = {
         const tagNameSegments = tagName.split('.').length;
         if (tagNameSegments === 1) {
           // Check for single identifier, like 'gql'
-          if (node.tag.type === 'Identifier' && node.tag.name !== tagName) {
+          if (node.tag.type !== 'Identifier' || node.tag.name !== tagName) {
             return;
           }
         } else if (tagNameSegments === 2){
```

Comment:
This currently doesn't have a case for `tagNameSegments === 3`. I'm not sure exactly how that would be implemented, but since this is related it might make sense to add the check in this PR? Even if to explicitly fail out?